lib/timers.js

```diff
@@ -430,36 +471,25 @@ function ontimeout(timer) {
     Reflect.apply(timer._onTimeout, timer, args);
 }
 
-function rearm(timer, start = TimerWrap.now()) {
+function rearm(timer, start) {
   // // Do not re-arm unenroll'd or closed timers.
-  if (timer._idleTimeout === -1) return;
+  if (timer._idleTimeout === -1)
+    return;
 
-  // If timer is unref'd (or was - it's permanently removed from the list.)
-  if (timer._handle && timer instanceof Timeout) {
-    timer._handle.start(timer._repeat);
-  } else {
-    timer._idleTimeout = timer._repeat;
+  const msecs = timer._idleTimeout = timer._repeat;
 
-    const duration = TimerWrap.now() - start;
-    if (duration >= timer._repeat) {
-      // If callback duration >= timer._repeat,
-      // add 1 ms to avoid blocking eventloop
-      insert(timer, false, start + duration - timer._repeat + 1);
-    } else {
-      insert(timer, false, start);
-    }
-  }
+  const now = TimerWrap.now();
+  if (start !== undefined && (now - start) >= msecs)
+    insert(timer, timer[kRefed], now - msecs + 1);
```

Comment:
Please keep the comment (and / or elaborate on it).

Reply:
While writing a more detailed comment for this I realized it's no longer necessary. Since we now always schedule the next run for at least 1ms, we can set the `start` to the correct value in every situation. This is also less likely to cause ordering inconsistencies.